changebot/github/github_api.py

```diff
@@ -99,6 +104,50 @@ def get_file_contents(self, path_to_file):
         contents_base64 = response.json()['content']
         return base64.b64decode(contents_base64).decode()
 
+    def get_user_config(self, path_to_file='.astropybot.yml',
+                        warn_on_failure=True):
+        """
+        Load user configuration for bot.
+
+        Parameters
+        ----------
+        path_to_file : str
+            Configuration file in YAML format in the repository.
+            If not given or invalid, default is used.
+
+        warn_on_failure : bool
+            Emit warning on failure to load YAML file.
+
+        Returns
+        -------
+        cfg : dict
+            Configuration parameters.
+
+        """
+        try:
+            file_content = self.get_file_contents(path_to_file)
+            cfg = yaml.load(file_content)
+        except Exception as e:
+            if warn_on_failure:
+                warnings.warn(str(e))
+            # Empty dict means calling code set the default
```

Comment:
I'm a bit worried about the behavior here - if there is a file but it doesn't parse as valid YAML, I think we just want to raise an error and stop the execution rather than continue with the default settings. If there is no file then we should go for the defaults.